eth/downloader/downloader.go

```diff
@@ -110,6 +113,11 @@ func New(mux *event.TypeMux, hasBlock hashCheckFn, getBlock getBlockFn) *Downloa
 		hashCh:    make(chan hashPack, 1),
 		blockCh:   make(chan blockPack, 1),
 	}
+	// Inject all the known bad hashes
+	downloader.banned = make(map[common.Hash]struct{})
```

Comment:
Use `set.Set` instead of raw map. Has better syntax. Use the `set.NewNonTS` one since it doesn't require a mutex.